### PR TITLE
Generate --all a different way

### PR DESCRIPTION
**Pull request by Insti**
Testing pain tends to be a sign that one's abstractions may be sub-optimal.

I tried a different approach by returning and iterating over an array of generators rather than calling something that also needed to construct them.

I think it turned out quite well.

What do you think?

Ref: https://github.com/exercism/xruby/pull/620

### CODE REVIEW

**Comment by Insti** [comment]
The revert at the start makes the diff look a bit noisy, try just viewing all but the first commit.

File: lib/generator/command_line.rb [hunk 1]
```diff
@@ -11,13 +11,13 @@ def initialize(paths)
     def parse(args)
       parser = GeneratorOptparser.new(args, @paths)
       @options = parser.options
-      generator if parser.options_valid?
+      return generators if parser.options_valid?
```
Comment (kotp) [inline lib/generator/command_line.rb]: I realize why the return is here, but with the plural form, this is clear even without the return explicitly there.

**Comment by Insti** [comment]
Refactoring of generator cases should probably be a separate PR but I am ok with sneaking it into this one if you don't feel like pulling them apart.
(Although it's just these 3 +  'clean up #available', so it should be easy to separate them.)

File: lib/generator/template_values.rb [hunk 2]
```diff
@@ -45,7 +45,7 @@ def extractor
     end
 
     def cases_load_name
-      Files::GeneratorCases.load_filename(paths.track, exercise_name)
+      Files::GeneratorCases.source_filename(paths.track, exercise_name)
```
Comment (hilary) [inline lib/generator/template_values.rb:48]: What is this change doing here? This isn't part of the changes needed.

**Comment by Insti** [comment]
Closing old PR.

**Pull request closed without merging**